### PR TITLE
BB2-835: Ansible variable cleanup

### PR DESCRIPTION
**Pull request by njdister**
**JIRA Ticket:**
[BB2-835](https://jira.cms.gov/browse/BB2-835)

**User Story or Bug Summary:**
There is a massive amount of unused variable entires in the ansible shared variable files as well as the vault due to the historical debt of the original designing team who used ansible to deploy instead of terraform (in addition to many planned but unimplemented features).  We should clean this up to reduce the cognitive load of anyone who might need to implement something new in these areas, such as in BB2-56.

### What Does This PR Do?

This PR takes the following approach to cleaning up variables:
- Ensure no dependencies within bluebutton-web-server repo by cross referencing (aka grepping) and then removing unneeded variables in .env file
- Then move to removing unneeded variables from all_var.yml and common.yml, which aggregate variables from environments
- Then move to removing unneeded variables the env.yml files from each environment
- Remove a couple files entirely not needed anymore (ca_env.yml and empty.yml)

I will save the actual vault cleanup for a separate PR to reduce the burden of review and the merge complexity.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check these things, in particular:

* Am I removing anything that is actually used?  I have fully tested this in the test environment, and intend to test this in sandbox and prod via canaries to be safe.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then here's a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence: N/A.
    * Does this PR add any new software dependencies? **No**.
    * Does this PR modify or invalidate any of our security controls? **No**.
    * Does this PR store or transmit data that was not stored or transmitted before? **No**.
* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR should not be merged unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons? **No**.

### What Needs to Be Merged and Deployed Before this PR?

Should coordinate merge with #1444 

### Submitter Checklist

I have gone through and verified that...:

* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] I have named this PR and its branch such that they'll be automatically be linked to the (most) relevant Jira issue, per: <https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html>.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. BFD, SLS, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: roles/env_vars/templates/env.j2 [hunk 1]
```diff
@@ -59,15 +50,9 @@ DJANGO_SLSX_VERIFY_SSL_INTERNAL={{ slsx_verify_ssl_internal }}
...
 # Misc. settings
-DJANGO_BLOCK_HTTP_REDIRECT_URIS="{{ django_block_http_redirect }}"
-DJANGO_OAUTH2_PROVIDER="{{ django_oauth2_provider }}"
 DJANGO_LOGGING="{{ django_logging }}"
-DJANGO_NO_UI="{{ django_no_ui }}"
 CORS_ORIGIN_ALLOW_ALL="{{ django_cors_origin_allow_all }}"
 
-# Tealim config
-{% if env in ["prod", "impl"] %}TEALIUM_ENV="prod"{% endif %}
-
```
Comment (njdister) [inline roles/env_vars/templates/env.j2:70]: Each removed variable here was carefully grepped for throughout the bluebutton-web-server repo to ensure no reliance.  The variables were either not used at all, or set by default in base.py and never pulled from the .env file.

File: vars/all_var.yml [hunk 2]
```diff
@@ -1,70 +1,11 @@
...
-
   ### DJANGO_SECRET_KEY:
-  # 8 Character Prefix and Suffix
-  # Relate Suffix to Deployment Environment
-  secret_pfx: "{{ env_secret_pfx }}"
-  secret_sfx: "{{ env_secret_sfx }}"
   # 50 character Random Key
   # http://www.miniwebtool.com/django-secret-key-generator/
-  rand_secret: "{{ env_rand_secret }}"
   django_secret_key: "{{ env_django_secret_key }}"
```
Comment (njdister) [inline vars/all_var.yml:9]: Important to note that secret_pfx, secret_sfx, and rand_secret are used only in the individual env.yml files to construct the final django_secret_key, so only django_secret_key is needed in the all vars file.

File: vars/all_var.yml [hunk 3]
```diff
@@ -96,38 +33,11 @@
   # False in IMPL / PROD
   django_debug: "{{ env_django_debug }}"
 
-  # The value of your django project's STATIC_ROOT settings.
-  # This will be the directory that django's `collectstatic` management command
-  # copies static files to, and it must be an absolute path. The default value
-  # here assumes that STATIC_ROOT is set in your settings.py like so:
-  # STATIC_ROOT = os.path.abspath(os.path.join(BASE_DIR, 'static'))
-  static_root: "{{ install_root }}/{{ project_name }}/{{ env_static_root_dir }}"
-
```
Comment (njdister) [inline vars/all_var.yml:105]: This is hardcoded in base.py settings and not used from here at all.
https://github.com/CMSgov/bluebutton-web-server/blob/7dbb1a65532c56a905bc825d6a7f170bf6f311c8/hhs_oauth_server/settings/base.py#L295

File: vars/all_var.yml [hunk 4]
```diff
@@ -96,38 +33,11 @@
...
   # application owner: pyapps
   app_owner: "pyapps"
   # application group: nginx
   app_group: "www-data"
 
-  ### External Internet Domain Name for the platform
-  external_dns_name: "{{ env_external_dns_name }}"
```
Comment (njdister) [inline vars/all_var.yml:112]: Duplicate unused variable, only actually used by reference to host_url below in playbooks.

File: vars/all_var.yml [hunk 5]
```diff
@@ -137,25 +47,10 @@
...
-  # django_email_backend:  'django.core.mail.backends.smtp.EmailBackend'
-  # django_email_backend:  'django.core.mail.backends.filebased.EmailBackend'
-  # django_email_backend:  'django.core.mail.backends.locmem.EmailBackend'
-  # django_email_backend:  'django.core.mail.backends.dummy.EmailBackend'
-  # django_email_backend:  'django_ses.SESBackend'
-
   django_email_backend: "{{ env_django_email_backend }}"
-
-  django_email_region: "{{ env_django_email_region }}"
-  django_email_ses_endpoint: "{{ env_django_email_ses_endpoint }}"
```
Comment (njdister) [inline vars/all_var.yml:150]: These are inherited/configured from the AWS instance role and not needed here.

File: vars/all_var.yml [hunk 6]
```diff
@@ -210,56 +92,12 @@
   db_user_name: "{{ env_db_user_name }}"
   db_user_pw: "{{ env_db_user_pw }}"
   db_platform_type: "postgres"
-  db_engine_driver: "django.db.backends.postgresql_psycopg2"
```
Comment (njdister) [inline vars/all_var.yml:213]: This value is not needed to construct the database connection string, or used anywhere else.

File: vars/all_var.yml [hunk 7]
```diff
@@ -210,56 +92,12 @@
...
   db_host_address: "db.bluebutton-{{ env }}.local"
   db_host_port: "15432"
   db_host_options: "?sslmode=require&options=-c role={{ env_db_role }}"
 
-  # Database SSL Certificate and Key Files:
-  ca_key: "server.key"
-  ca_cert: "server.crt"
-
+  # legacy value still used in roles, should be renamed for clarity
   cf_region: "{{ aws_region }}"
```
Comment (njdister) [inline vars/all_var.yml:100]: Unfortunately this variable is still used in playbooks/roles, opted to leave it and not refactor for now.

File: vars/all_var.yml [hunk 8]
```diff
@@ -273,97 +111,21 @@
...
-  cf_app_key_name: "{{ env_cf_app_key_name }}"
-  cf_app_instance_type: "{{ env_cf_app_instance_type }}"
-  cf_app_public_server_name: "{{ env_cf_app_public_server_name }}"
-  cf_app_pyserver_name: "{{ env_cf_app_pyserver_name }}"
-  cf_app_pyapp_settings: "{{ env_cf_app_pyapp_settings }}"
-  cf_app_repo_branch: "{{ env_cf_app_repo_branch }}"
-  cf_app_db_user: "{{ env_cf_app_db_user }}"
-  cf_app_db_pwd: "{{ env_cf_app_db_pwd }}"
-  cf_app_pyapps_email: "{{ env_cf_app_pyapps_email }}"
   cf_app_pyapps_pwd: "{{ env_cf_app_pyapps_pwd }}"
```
Comment (njdister) [inline vars/all_var.yml:116]: Unfortunately these variables are still used in playbooks/roles, opted to leave it and not refactor for now.

File: vars/all_var.yml [hunk 9]
```diff
@@ -273,97 +111,21 @@
...
-
-  app_lb_arn: "{{ env_app_lb_arn}}"
-  app_ssl_lb_arn: "{{ env_app_ssl_lb_arn }}"
-  app_lb_app_protocol: "https"
-
   # splunk configuration
   # Group name = BBAPI
   splunk_group_name: "{{ env_splunk_group_name }}"
   splunk_target_layer: "{{ env_splunk_target_layer }}"
-  splunk_client_name: "{{ env_splunk_client_name }}"
```
Comment (njdister) [inline vars/all_var.yml:338]: The splunk client name is constructed from all other values directly in the splunk playbooks/roles so this value is never used from here.

File: vars/all_var.yml [hunk 10]
```diff
@@ -379,19 +141,7 @@
...
-
-  django_oauth2_provider: "{{ env_django_oauth2_provider }}"
-
-  # file to append to in /var/log/pyapps/
-  django_export_model_file: "export.log"
-
   # new relic
   new_relic_license_key: "{{ vault_env_new_relic_license_key }}"
-
-  allowed_ips: "{{ vault_env_allowed_ips }}"
```
Comment (njdister) [inline vars/all_var.yml:397]: No longer needed since we moved all IP protection into Akamai.

**Review by JFU-GIT — APPROVED** [review]
thanks for the long overdue cleanup work

**Review by dtisza1 — APPROVED** [review]
This looks good to me!

I performed a few passes over the changes. Nothing sticks out to me as being a possible issue. Good work on this!

**Pull request merged**